dev/com.ibm.ws.security.javaeesec/src/com/ibm/ws/security/javaeesec/properties/ModulePropertiesUtils.java

```diff
@@ -88,7 +89,20 @@ public HttpAuthenticationMechanism getHttpAuthenticationMechanism() {
 
     @SuppressWarnings({ "unchecked", "rawtypes" })
     private HttpAuthenticationMechanism getHttpAuthenticationMechanism(boolean logError) {
-        MetaData metadata = getMetaData();
+        MetaData metadata = null;
+
+        // if we already know this web module doesn't have HAM, we can skip it
+        WebModuleMetaData wmmd = getWebModuleMetaData();
```

Comment:
```suggestion
        WebModuleMetaDataImpl wmmd = (WebModuleMetaDataImpl) getWebModuleMetaData();
```

Reply:
Done

Reply:
You didn't actually do this.  You said it wouldn't work for you, but it has to in order to compile for real.  It should cast up front here and then remove the extra casts below.

Reply:
Yup, my bad ... this time I've _really done it_